### ISSUE TITLE
Sectioned list is not updated correctly

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#3502]
### 👥 Impact
This is the most frequently occurring bug in 1.12.1

### 🔗 Context
Probably related to #3287 

### 🛰 Link
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/2703bae32546d5893b514c77b10f73ba85f1018c?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/0a8216ac8d2ef3aff7ea003741de9dfb1c7e4d9c?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/194d4943c4e526d36c924b867a1bc6d22f52dc1b?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/168ef42743e6d2ae4cc45df5b50172e2186e924d?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/68a0435bd2fa28431b9ae4ff79fe825e346867ca?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/cb53e518fe6131cf8b037d79b8a752177ddf47e9?appVersion=1.12.1
- https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/2703bae32546d5893b514c77b10f73ba85f1018c?appVersion=1.12.1

### 📝  Exception & Stack Trace
```cs
ReactiveTableViewBinder`2[TModel,TCell].handleCollectionChange (Toggl.Foundation.MvvmCross.Collections.Changes.ICollectionChange change)
```

Different messages - some examples:
```
Invalid update: invalid number of sections. The number of sections contained in the table view after the update (41) must be equal to the number of sections contained in the table view before the update (43), plus or minus the number of sections inserted or deleted (1 inserted, 0 deleted).

Invalid update: invalid number of rows in section 0. The number of rows contained in an existing section after the update (11) must be equal to the number of rows contained in that section before the update (8), plus or minus the number of rows inserted or deleted from that section (0 inserted, 0 deleted) and plus or minus the number of rows moved into or out of that section (0 moved in, 0 moved out).

Reason: attempt to delete and reload the same index path (<NSIndexPath: 0x1c6824d40> {length = 2, path = 37 - 9223372036854775807})

Invalid update: invalid number of rows in section 56. The number of rows contained in an existing section after the update (24) must be equal to the number of rows contained in that section before the update (30), plus or minus the number of rows inserted or deleted from that section (0 inserted, 1 deleted) and plus or minus the number of rows moved into or out of that section (0 moved in, 0 moved out).
```

**Comment by simonrozsival** [comment]
This might very well be fixed by #3513 - we should check the crash log after the release and see if was resolved by it. If not, we can revisit this issue and release a fix in a hotfix.

**Comment by heytherewill** [comment]
👴
